lightning-invoice/src/payment.rs

```diff
@@ -270,7 +270,7 @@ pub trait Payer {
 pub trait Router {
 	/// Finds a [`Route`] between `payer` and `payee` for a payment with the given values.
 	fn find_route(
-		&self, payer: &PublicKey, route_params: &RouteParameters, payment_hash: &PaymentHash,
+		&self, payer: &PublicKey, route_params: &RouteParameters,
```

Comment:
It wouldn't be ideal to provide users with the payment hash in trampoline payments. 

Do either of these options seem reasonable:
(1) make payment_hash into an `Option<(PaymentHash, PaymentId)>`
(2) make payment_hash into an `Option<PaymentId>`

Reply:
> It wouldn't be ideal to provide users with the payment hash in trampoline payments.
> 

Could you expand on this? Is the hash not available? Or is providing it harmful / misleading in some way?

Reply:
I think it's leaking the sender's privacy since it's not our payment

Reply:
> It wouldn't be ideal to provide users with the payment hash in trampoline payments.
> 
> Do either of these options seem reasonable: (1) make payment_hash into an `Option<(PaymentHash, PaymentId)>` (2) make payment_hash into an `Option<PaymentId>`

Where is `PaymentId` coming from / why is it needed? 

> I think it's leaking the sender's privacy since it's not our payment

Not really sure I follow this. If our node knows the payment hash, hasn't it already been leaked?

Reply:
IIUC there's still a difference in privacy between seeing it in an internal `update_add_htlc` message vs actually providing it to a user in a trait method. 

> Where is PaymentId coming from / why is it needed?

Same reason a `payment_hash` is being provided in 592bfd7c58e73eaa5cfd5032d982883878510037 I think, to have another option to look up pre-computed probes. As of #1761, `PaymentId`s will be required upon all payment initiations, so it'd come from there for non-trampoline `find_route` calls.